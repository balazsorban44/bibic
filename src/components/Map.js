import React from 'react'
import bibicBird from '../assets/icons/bibic-white.png'
import GoogleMaps from 'google-map-react'

import {MAPS_API_KEY, MAPS_STYLE } from '../lib/google-maps'

const Marker = () => <img className="bibic-marker" alt="" src={bibicBird}/>

const Map = () => (
  <React.Fragment>
    <GoogleMaps
      options={{
        styles: MAPS_STYLE
      }}
      bootstrapURLKeys={{ key: [MAPS_API_KEY] }}
      center={{
        lat: 46.3950025,
        lng: 17.506616
      }}
      zoom={15}
    >
      <Marker
        lat={46.3950025}
        lng={17.506616}
      />
    </GoogleMaps>
  </React.Fragment> 
)

export default Map