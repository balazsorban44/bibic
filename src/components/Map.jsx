import React from 'react'
import bibic from '../assets/icons/bibic-white.png'
import GoogleMaps from "google-map-react"

import {MAPS_API_KEY, MAPS_STYLE} from '../lib/google-maps'


export const Marker = () =>
  <img
    alt=""
    className="bibic-marker"
    src={bibic}
  />

export default () =>
  <div className="map">
    <GoogleMaps
      bootstrapURLKeys={{key: [MAPS_API_KEY]}}
      center={{lat: 46.3950025,
        lng: 17.506616}}
      options={{styles: MAPS_STYLE}}
      zoom={15}
    >
      <Marker
        lat={46.3950025}
        lng={17.506616}
      />
    </GoogleMaps>
  </div>