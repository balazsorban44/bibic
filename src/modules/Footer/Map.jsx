import React from "react"
import bibic from "assets/icons/bibic-white.png"
import GoogleMaps from "google-map-react"

import {MAPS_API_KEY, MAPS_STYLE} from "lib/google-maps"

const center = {
  lat: 46.3950025,
  lng: 17.506616
}

export const Marker = () =>
  <img
    alt=""
    className="bibic-marker"
    src={bibic}
  />

const Map = () =>
  <div className="map">
    <GoogleMaps
      bootstrapURLKeys={{key: [MAPS_API_KEY]}}
      center={center}
      options={{styles: MAPS_STYLE}}
      zoom={15}
    >
      <Marker
        lat={center.lat}
        lng={center.lng}
      />
    </GoogleMaps>
  </div>

export default Map