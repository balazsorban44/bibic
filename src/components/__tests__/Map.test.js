import Map, {Marker} from "../Map"

describe("Map component", () => {
  const props = {}
  const wrapper = shallow(<Map {...props}/>)

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot()
  })
})

describe("Marker component", () => {
  const wrapper = shallow(<Marker/>)

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot()
  })
})