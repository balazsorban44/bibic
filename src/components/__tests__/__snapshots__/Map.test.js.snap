// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <GoogleMap
        bootstrapURLKeys={
          Object {
            "key": Array [
              "AIzaSyDvnPguJofFhvUbJiNYkR2rlzSvIeZhco8",
            ],
          }
        }
        center={
          Object {
            "lat": 46.3950025,
            "lng": 17.506616,
          }
        }
        debounced={true}
        distanceToMouse={[Function]}
        googleMapLoader={[Function]}
        heatmap={Object {}}
        heatmapLibrary={false}
        hoverDistance={30}
        layerTypes={Array []}
        options={
          Object {
            "styles": Array [
              Object {
                "elementType": "geometry",
                "featureType": "water",
                "stylers": Array [
                  Object {
                    "color": "#e9e9e9",
                  },
                  Object {
                    "lightness": 17,
                  },
                ],
              },
              Object {
                "elementType": "geometry",
                "featureType": "landscape",
                "stylers": Array [
                  Object {
                    "color": "#f5f5f5",
                  },
                  Object {
                    "lightness": 20,
                  },
                ],
              },
              Object {
                "elementType": "geometry.fill",
                "featureType": "road.highway",
                "stylers": Array [
                  Object {
                    "color": "#ffffff",
                  },
                  Object {
                    "lightness": 17,
                  },
                ],
              },
              Object {
                "elementType": "geometry.stroke",
                "featureType": "road.highway",
                "stylers": Array [
                  Object {
                    "color": "#ffffff",
                  },
                  Object {
                    "lightness": 29,
                  },
                  Object {
                    "weight": 0.2,
                  },
                ],
              },
              Object {
                "elementType": "geometry",
                "featureType": "road.arterial",
                "stylers": Array [
                  Object {
                    "color": "#ffffff",
                  },
                  Object {
                    "lightness": 18,
                  },
                ],
              },
              Object {
                "elementType": "geometry",
                "featureType": "road.local",
                "stylers": Array [
                  Object {
                    "color": "#ffffff",
                  },
                  Object {
                    "lightness": 16,
                  },
                ],
              },
              Object {
                "elementType": "geometry",
                "featureType": "poi",
                "stylers": Array [
                  Object {
                    "color": "#f5f5f5",
                  },
                  Object {
                    "lightness": 21,
                  },
                ],
              },
              Object {
                "elementType": "geometry",
                "featureType": "poi.park",
                "stylers": Array [
                  Object {
                    "color": "#dedede",
                  },
                  Object {
                    "lightness": 21,
                  },
                ],
              },
              Object {
                "elementType": "labels.text.stroke",
                "stylers": Array [
                  Object {
                    "visibility": "on",
                  },
                  Object {
                    "color": "#ffffff",
                  },
                  Object {
                    "lightness": 16,
                  },
                ],
              },
              Object {
                "elementType": "labels.text.fill",
                "stylers": Array [
                  Object {
                    "saturation": 36,
                  },
                  Object {
                    "color": "#333333",
                  },
                  Object {
                    "lightness": 40,
                  },
                ],
              },
              Object {
                "elementType": "labels.icon",
                "stylers": Array [
                  Object {
                    "visibility": "off",
                  },
                ],
              },
              Object {
                "elementType": "geometry",
                "featureType": "transit",
                "stylers": Array [
                  Object {
                    "color": "#f2f2f2",
                  },
                  Object {
                    "lightness": 19,
                  },
                ],
              },
              Object {
                "elementType": "geometry.fill",
                "featureType": "administrative",
                "stylers": Array [
                  Object {
                    "color": "#fefefe",
                  },
                  Object {
                    "lightness": 20,
                  },
                ],
              },
              Object {
                "elementType": "geometry.stroke",
                "featureType": "administrative",
                "stylers": Array [
                  Object {
                    "color": "#fefefe",
                  },
                  Object {
                    "lightness": 17,
                  },
                  Object {
                    "weight": 1.2,
                  },
                ],
              },
            ],
          }
        }
        style={
          Object {
            "height": "100%",
            "margin": 0,
            "padding": 0,
            "position": "relative",
            "width": "100%",
          }
        }
        yesIWantToUseGoogleMapApiInternals={false}
        zoom={15}
      >
        <Marker
          lat={46.3950025}
          lng={17.506616}
        />
      </GoogleMap>,
      "className": "map",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bootstrapURLKeys": Object {
          "key": Array [
            "AIzaSyDvnPguJofFhvUbJiNYkR2rlzSvIeZhco8",
          ],
        },
        "center": Object {
          "lat": 46.3950025,
          "lng": 17.506616,
        },
        "children": <Marker
          lat={46.3950025}
          lng={17.506616}
        />,
        "debounced": true,
        "distanceToMouse": [Function],
        "googleMapLoader": [Function],
        "heatmap": Object {},
        "heatmapLibrary": false,
        "hoverDistance": 30,
        "layerTypes": Array [],
        "options": Object {
          "styles": Array [
            Object {
              "elementType": "geometry",
              "featureType": "water",
              "stylers": Array [
                Object {
                  "color": "#e9e9e9",
                },
                Object {
                  "lightness": 17,
                },
              ],
            },
            Object {
              "elementType": "geometry",
              "featureType": "landscape",
              "stylers": Array [
                Object {
                  "color": "#f5f5f5",
                },
                Object {
                  "lightness": 20,
                },
              ],
            },
            Object {
              "elementType": "geometry.fill",
              "featureType": "road.highway",
              "stylers": Array [
                Object {
                  "color": "#ffffff",
                },
                Object {
                  "lightness": 17,
                },
              ],
            },
            Object {
              "elementType": "geometry.stroke",
              "featureType": "road.highway",
              "stylers": Array [
                Object {
                  "color": "#ffffff",
                },
                Object {
                  "lightness": 29,
                },
                Object {
                  "weight": 0.2,
                },
              ],
            },
            Object {
              "elementType": "geometry",
              "featureType": "road.arterial",
              "stylers": Array [
                Object {
                  "color": "#ffffff",
                },
                Object {
                  "lightness": 18,
                },
              ],
            },
            Object {
              "elementType": "geometry",
              "featureType": "road.local",
              "stylers": Array [
                Object {
                  "color": "#ffffff",
                },
                Object {
                  "lightness": 16,
                },
              ],
            },
            Object {
              "elementType": "geometry",
              "featureType": "poi",
              "stylers": Array [
                Object {
                  "color": "#f5f5f5",
                },
                Object {
                  "lightness": 21,
                },
              ],
            },
            Object {
              "elementType": "geometry",
              "featureType": "poi.park",
              "stylers": Array [
                Object {
                  "color": "#dedede",
                },
                Object {
                  "lightness": 21,
                },
              ],
            },
            Object {
              "elementType": "labels.text.stroke",
              "stylers": Array [
                Object {
                  "visibility": "on",
                },
                Object {
                  "color": "#ffffff",
                },
                Object {
                  "lightness": 16,
                },
              ],
            },
            Object {
              "elementType": "labels.text.fill",
              "stylers": Array [
                Object {
                  "saturation": 36,
                },
                Object {
                  "color": "#333333",
                },
                Object {
                  "lightness": 40,
                },
              ],
            },
            Object {
              "elementType": "labels.icon",
              "stylers": Array [
                Object {
                  "visibility": "off",
                },
              ],
            },
            Object {
              "elementType": "geometry",
              "featureType": "transit",
              "stylers": Array [
                Object {
                  "color": "#f2f2f2",
                },
                Object {
                  "lightness": 19,
                },
              ],
            },
            Object {
              "elementType": "geometry.fill",
              "featureType": "administrative",
              "stylers": Array [
                Object {
                  "color": "#fefefe",
                },
                Object {
                  "lightness": 20,
                },
              ],
            },
            Object {
              "elementType": "geometry.stroke",
              "featureType": "administrative",
              "stylers": Array [
                Object {
                  "color": "#fefefe",
                },
                Object {
                  "lightness": 17,
                },
                Object {
                  "weight": 1.2,
                },
              ],
            },
          ],
        },
        "style": Object {
          "height": "100%",
          "margin": 0,
          "padding": 0,
          "position": "relative",
          "width": "100%",
        },
        "yesIWantToUseGoogleMapApiInternals": false,
        "zoom": 15,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "lat": 46.3950025,
          "lng": 17.506616,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <GoogleMap
          bootstrapURLKeys={
            Object {
              "key": Array [
                "AIzaSyDvnPguJofFhvUbJiNYkR2rlzSvIeZhco8",
              ],
            }
          }
          center={
            Object {
              "lat": 46.3950025,
              "lng": 17.506616,
            }
          }
          debounced={true}
          distanceToMouse={[Function]}
          googleMapLoader={[Function]}
          heatmap={Object {}}
          heatmapLibrary={false}
          hoverDistance={30}
          layerTypes={Array []}
          options={
            Object {
              "styles": Array [
                Object {
                  "elementType": "geometry",
                  "featureType": "water",
                  "stylers": Array [
                    Object {
                      "color": "#e9e9e9",
                    },
                    Object {
                      "lightness": 17,
                    },
                  ],
                },
                Object {
                  "elementType": "geometry",
                  "featureType": "landscape",
                  "stylers": Array [
                    Object {
                      "color": "#f5f5f5",
                    },
                    Object {
                      "lightness": 20,
                    },
                  ],
                },
                Object {
                  "elementType": "geometry.fill",
                  "featureType": "road.highway",
                  "stylers": Array [
                    Object {
                      "color": "#ffffff",
                    },
                    Object {
                      "lightness": 17,
                    },
                  ],
                },
                Object {
                  "elementType": "geometry.stroke",
                  "featureType": "road.highway",
                  "stylers": Array [
                    Object {
                      "color": "#ffffff",
                    },
                    Object {
                      "lightness": 29,
                    },
                    Object {
                      "weight": 0.2,
                    },
                  ],
                },
                Object {
                  "elementType": "geometry",
                  "featureType": "road.arterial",
                  "stylers": Array [
                    Object {
                      "color": "#ffffff",
                    },
                    Object {
                      "lightness": 18,
                    },
                  ],
                },
                Object {
                  "elementType": "geometry",
                  "featureType": "road.local",
                  "stylers": Array [
                    Object {
                      "color": "#ffffff",
                    },
                    Object {
                      "lightness": 16,
                    },
                  ],
                },
                Object {
                  "elementType": "geometry",
                  "featureType": "poi",
                  "stylers": Array [
                    Object {
                      "color": "#f5f5f5",
                    },
                    Object {
                      "lightness": 21,
                    },
                  ],
                },
                Object {
                  "elementType": "geometry",
                  "featureType": "poi.park",
                  "stylers": Array [
                    Object {
                      "color": "#dedede",
                    },
                    Object {
                      "lightness": 21,
                    },
                  ],
                },
                Object {
                  "elementType": "labels.text.stroke",
                  "stylers": Array [
                    Object {
                      "visibility": "on",
                    },
                    Object {
                      "color": "#ffffff",
                    },
                    Object {
                      "lightness": 16,
                    },
                  ],
                },
                Object {
                  "elementType": "labels.text.fill",
                  "stylers": Array [
                    Object {
                      "saturation": 36,
                    },
                    Object {
                      "color": "#333333",
                    },
                    Object {
                      "lightness": 40,
                    },
                  ],
                },
                Object {
                  "elementType": "labels.icon",
                  "stylers": Array [
                    Object {
                      "visibility": "off",
                    },
                  ],
                },
                Object {
                  "elementType": "geometry",
                  "featureType": "transit",
                  "stylers": Array [
                    Object {
                      "color": "#f2f2f2",
                    },
                    Object {
                      "lightness": 19,
                    },
                  ],
                },
                Object {
                  "elementType": "geometry.fill",
                  "featureType": "administrative",
                  "stylers": Array [
                    Object {
                      "color": "#fefefe",
                    },
                    Object {
                      "lightness": 20,
                    },
                  ],
                },
                Object {
                  "elementType": "geometry.stroke",
                  "featureType": "administrative",
                  "stylers": Array [
                    Object {
                      "color": "#fefefe",
                    },
                    Object {
                      "lightness": 17,
                    },
                    Object {
                      "weight": 1.2,
                    },
                  ],
                },
              ],
            }
          }
          style={
            Object {
              "height": "100%",
              "margin": 0,
              "padding": 0,
              "position": "relative",
              "width": "100%",
            }
          }
          yesIWantToUseGoogleMapApiInternals={false}
          zoom={15}
        >
          <Marker
            lat={46.3950025}
            lng={17.506616}
          />
        </GoogleMap>,
        "className": "map",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bootstrapURLKeys": Object {
            "key": Array [
              "AIzaSyDvnPguJofFhvUbJiNYkR2rlzSvIeZhco8",
            ],
          },
          "center": Object {
            "lat": 46.3950025,
            "lng": 17.506616,
          },
          "children": <Marker
            lat={46.3950025}
            lng={17.506616}
          />,
          "debounced": true,
          "distanceToMouse": [Function],
          "googleMapLoader": [Function],
          "heatmap": Object {},
          "heatmapLibrary": false,
          "hoverDistance": 30,
          "layerTypes": Array [],
          "options": Object {
            "styles": Array [
              Object {
                "elementType": "geometry",
                "featureType": "water",
                "stylers": Array [
                  Object {
                    "color": "#e9e9e9",
                  },
                  Object {
                    "lightness": 17,
                  },
                ],
              },
              Object {
                "elementType": "geometry",
                "featureType": "landscape",
                "stylers": Array [
                  Object {
                    "color": "#f5f5f5",
                  },
                  Object {
                    "lightness": 20,
                  },
                ],
              },
              Object {
                "elementType": "geometry.fill",
                "featureType": "road.highway",
                "stylers": Array [
                  Object {
                    "color": "#ffffff",
                  },
                  Object {
                    "lightness": 17,
                  },
                ],
              },
              Object {
                "elementType": "geometry.stroke",
                "featureType": "road.highway",
                "stylers": Array [
                  Object {
                    "color": "#ffffff",
                  },
                  Object {
                    "lightness": 29,
                  },
                  Object {
                    "weight": 0.2,
                  },
                ],
              },
              Object {
                "elementType": "geometry",
                "featureType": "road.arterial",
                "stylers": Array [
                  Object {
                    "color": "#ffffff",
                  },
                  Object {
                    "lightness": 18,
                  },
                ],
              },
              Object {
                "elementType": "geometry",
                "featureType": "road.local",
                "stylers": Array [
                  Object {
                    "color": "#ffffff",
                  },
                  Object {
                    "lightness": 16,
                  },
                ],
              },
              Object {
                "elementType": "geometry",
                "featureType": "poi",
                "stylers": Array [
                  Object {
                    "color": "#f5f5f5",
                  },
                  Object {
                    "lightness": 21,
                  },
                ],
              },
              Object {
                "elementType": "geometry",
                "featureType": "poi.park",
                "stylers": Array [
                  Object {
                    "color": "#dedede",
                  },
                  Object {
                    "lightness": 21,
                  },
                ],
              },
              Object {
                "elementType": "labels.text.stroke",
                "stylers": Array [
                  Object {
                    "visibility": "on",
                  },
                  Object {
                    "color": "#ffffff",
                  },
                  Object {
                    "lightness": 16,
                  },
                ],
              },
              Object {
                "elementType": "labels.text.fill",
                "stylers": Array [
                  Object {
                    "saturation": 36,
                  },
                  Object {
                    "color": "#333333",
                  },
                  Object {
                    "lightness": 40,
                  },
                ],
              },
              Object {
                "elementType": "labels.icon",
                "stylers": Array [
                  Object {
                    "visibility": "off",
                  },
                ],
              },
              Object {
                "elementType": "geometry",
                "featureType": "transit",
                "stylers": Array [
                  Object {
                    "color": "#f2f2f2",
                  },
                  Object {
                    "lightness": 19,
                  },
                ],
              },
              Object {
                "elementType": "geometry.fill",
                "featureType": "administrative",
                "stylers": Array [
                  Object {
                    "color": "#fefefe",
                  },
                  Object {
                    "lightness": 20,
                  },
                ],
              },
              Object {
                "elementType": "geometry.stroke",
                "featureType": "administrative",
                "stylers": Array [
                  Object {
                    "color": "#fefefe",
                  },
                  Object {
                    "lightness": 17,
                  },
                  Object {
                    "weight": 1.2,
                  },
                ],
              },
            ],
          },
          "style": Object {
            "height": "100%",
            "margin": 0,
            "padding": 0,
            "position": "relative",
            "width": "100%",
          },
          "yesIWantToUseGoogleMapApiInternals": false,
          "zoom": 15,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "lat": 46.3950025,
            "lng": 17.506616,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Marker component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Marker />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "alt": "",
      "className": "bibic-marker",
      "src": "bibic-white.png",
    },
    "ref": null,
    "rendered": null,
    "type": "img",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "",
        "className": "bibic-marker",
        "src": "bibic-white.png",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
